### ISSUE TITLE
Drawer: Duplicate role="dialog" attributes in Drawer

### ISSUE THREAD

**Issue by ywzheng1** [grafana/grafana#112678]
### What happened?

The `Drawer` component currently renders two nested elements with role="dialog", which violates accessibility best practices. 

<img width="712" height="345" alt="Image" src="https://github.com/user-attachments/assets/2d3b1445-e819-4a7d-8098-8d26f54935fc" />

### What did you expect to happen?

- Only one `role="dialog"` should be present
- The inner element should not have a dialog role. It can remain as a div or use role="presentation" if needed.

### Did this work before?

Not sure

### How do we reproduce it?

1. Open any drawer element on FE
2. Inspect element `class="rc-drawer-content"` and inner element
3. Two `role="dialog"` appears

### Environment (with versions)?

Grafana: v12.3.0-pre
OS: Mac
Browser: Chrome

### Grafana platform?

None
